### PR TITLE
[ci] Switch to self-hosted PyPI for nightly release

### PR DESCRIPTION
**Pull request by qiao-bo**
We will use self-hosted pypi for nightly release. This PR updates CI and Docs.

### CODE REVIEW

File: misc/upload_release.py [hunk 1]
```diff
@@ -45,9 +45,11 @@ def upload_artifact(is_taichi):
         sys.exit(f'Missing password env var {pwd_env}')
     command = [sys.executable, '-m', 'twine', 'upload']
     if not is_taichi:
-        command.extend(['--repository', 'testpypi'])
-    command.extend(
-        ['--verbose', '-u', '__token__', '-p', twine_password, 'dist/*'])
+        command.extend(
+            ['--repository-url', 'https://pypi.taichi.graphics/simple/'])
+    uname = '__token__' if is_taichi else 'admin'
```
Comment (frostming) [inline misc/upload_release.py:50]: Does pypicloud supports tokens? If so, we should better use a token.

Comment (qiao-bo) [reply misc/upload_release.py:50]: @frostming: @rexwangcc and I just checked that Pypi Cloud seems not have access token support (there is registration token though, which does not help). Should we just stick to the user name login?

Comment (rexwangcc) [reply misc/upload_release.py:50]: Based on what we have learned from the pypicloud setup, I suggest we stick the basic auth login approach (of course we are using a non-admin account for uploading) :P

File: README.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ pip install taichi
 *We also provide a nightly package. Note that nighly packages may crash because they are not fully tested.  We cannot guarantee their validity, and you are at your own risk trying out our latest, untested features.*
 
 ```bash
-pip install -i https://test.pypi.org/simple/ taichi-nightly
+pip install -i https://pypi.taichi.graphics/simple/ taichi-nightly
```
Comment (rexwangcc) [inline README.md:76]: nit: would be better to add a comment above this line briefly explain why we are using our self-hosted PyPI for nightly?

File: misc/upload_release.py (hunk 1 above)
Comment (rexwangcc) [reply misc/upload_release.py:50]:
```suggestion
    uname = '__token__' if is_taichi else 'taichi-nightly-cloud-pypi-uploader'
```

**Review by rexwangcc — APPROVED** [review]
I have one comment otherwise LGTM, thanks!

**Review by frostming — APPROVED** [review]

**Pull request merged**